sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDD.scala

```diff
@@ -29,14 +29,51 @@ import org.apache.spark.sql.internal.SessionState
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.util.SerializableConfiguration
 
+// This doesn't directly override RDD methods as MiMa complains it.
+abstract class BaseStateStoreRDD[T: ClassTag, U: ClassTag](
+    dataRDD: RDD[T],
+    checkpointLocation: String,
+    queryRunId: UUID,
+    operatorId: Long,
+    sessionState: SessionState,
+    @transient private val storeCoordinator: Option[StateStoreCoordinatorRef],
+    extraOptions: Map[String, String] = Map.empty) extends RDD[U](dataRDD) {
+
+  protected val storeConf = new StateStoreConf(sessionState.conf, extraOptions)
+
+  // A Hadoop Configuration can be about 10 KB, which is pretty big, so broadcast it
+  protected val hadoopConfBroadcast = dataRDD.context.broadcast(
+    new SerializableConfiguration(sessionState.newHadoopConf()))
+
+  /** Implementations can simply call this method in getPreferredLocations. */
+  protected def _getPartitions: Array[Partition] = dataRDD.partitions
```

Comment:
Not yet see why don't we just provide a default implementation. Could you elaborate?

Reply:
As I commented in the class doc - MiMa somehow complained it. Not 100% sure why.

Reply:
Just seen the MiMa comment at the top of the class. Spent some time to find out the cause but as I see it would lead far and don't really see the return of the investment. I'm fine with this.

Reply:
Does mima complaining refers to this error?
```
[error]  * abstract method getPartitions()Array[org.apache.spark.Partition] in class org.apache.spark.rdd.RDD does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectAbstractMethodProblem]("org.apache.spark.rdd.RDD.getPartitions")
```
Maybe we can add a filter since it's not a public API change. I'll also take a further look at this error message.

Reply:
Filter to what exactly? Can you give an example?

Reply:
I wouldn't want to touch RDD side just because of this change, which may bring unnecessary additional discussion. I don't see this as a kind of blocker for this PR. Anyone can deal with this as a follow-up, with possibility of discussion.

Reply:
```
Filter to what exactly? Can you give an example?
```
Sorry for the unclear, I mean add a filter in `./project/MimaExcludes.scala`

```
I wouldn't want to touch RDD side just because of this change
```
Yeah, of cause we don't want to touch RDD

Reply:
That was what I understand and I wouldn’t do it in this PR. What we need to add based on the error message is to exclude the some pattern against RDD, which I’m not 100% sure everyone would agree with. That’s what I said “additional discussion”.

Reply:
I don’t mean I don’t like to change, I mean this shouldn’t be blocked any longer by something which can be done as a follow-up. If you’d like to volunteer to look into the issue I’m happy to, “as a follow-up PR”.

Reply:
Thanks for the further explanation, agree to do it in a follow-up.